tools/search_data_parser/parser.py

```diff
@@ -0,0 +1,63 @@
+from dataclasses import dataclass
+import json
+import os
+import argparse
+
+@dataclass
+class Course:
+    _id: str
+    subject: str
+    title: str
+    code: str
+    instructors: list
+
+def main(seed_path: str, out_path: str):
+    course_map: dict[str, Course] = {}
+
+    for filename in sorted(os.listdir(seed_path)):
+        file_path = os.path.join(seed_path, filename)
+
+        if not os.path.isfile(file_path) or "course" not in file_path:
+            continue
+
+        with open(file_path, "r") as fobj:
+            courses = json.load(fobj)
+            for course in courses:
+                course_id = course["_id"]
+                if course_id not in course_map:
+                    course_map[course_id] = Course(
+                        course["_id"],
+                        course["subject"],
+                        course["title"],
+                        course["code"],
+                        course["instructors"],
+                    )
+                else:
+                    course_map[course_id].instructors += course["instructors"]
+
+    output = [course.__dict__ for course in course_map.values()]
+
+    with open(out_path, "w") as f:
+        json.dump(output, f, indent=2)
```

Comment:
`indent=2` makes the file a lot larger than needed, since it adds a ton of newlines.
The file doesn't have to be human readable so you should remove this to minify the JSON.
Getting rid of this reduces it from ~20mb -> ~14mb so it's quite substantial

Reply:
good point, let me try some stuff out to reduce bundle size

Reply:
The compression thing I mentioned probably isn't that relevant actually, since in prod I think it's already compressed, so we should just focus on getting the raw file size down

Reply:
@39bytes wanna take a look at this new structure? its 1.5MB locally for me 

```json
{
  "courses": Course
  "instructors": string (name) 
}
```

Reply:
Much better, we can still minify the JSON even more though by stripping all of the whitespace.
I'm just gonna make that change and also rename directory to match other things in `tools`